modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx

```diff
@@ -52,133 +86,174 @@ handled by the codemod are marked with 🤖 in the Upgrade Guide.**
-for ideas on how to replace the deprecated `Layout` component with the `Grid` component. This will
-provide you with a deeper understanding of how to set custom column width or positioning using the
-`Grid` component.
+We've removed the `Stack`, `HStack`, and `VStack` components from Canvas Kit (for reference, see
+[`Stack`](https://d2krrudi3mmzzw.cloudfront.net/v8/?path=/docs/components-layout-stack--basic-stack)
+from v8). We recommend using [`Flex`](/components/layout/flex/) instead. `Flex` supports the same
+consistent spacing between its elements via the `gap` prop (analogous to the `spacing` prop from
+`Stack`).
 
----
+```tsx
+// v8
+<Stack spacing="s">
+  ...
+</Stack>
 
-### Stack, HStack and vstack
+// v9
+<Flex gap="s">
+  ...
+</Flex>
+```
 
-We've hard deprecated the `Stack`, `HStack` and `VStack` components along with the props associated
-to those components (`StackProps`, `HStackProps`, `VStackProps`, `StackStyleProps`). We recommend
-using the `Flex` component and to replace the `spacing` prop with the `gap` prop. You can also run
-`npx @workday/canvas-kit-codemod v9 [path]`.
+The orientation of `VStack` elements can be replicated with `Flex` using `flexDirection`.
 
-> **Note**: If you are consuming `StackProps` for a custom component, please make sure to change the
-> `spacing` prop to the `gap` prop
+```tsx
+// v8
+<VStack>
+  ...
+</VStack>
+
+// v9
+<Flex flexDirection="column">
+  ...
+</Flex>
+```
+
+The `StackProps`, `HStackProps`, `VStackProps`, and `StackStyleProps` types have been removed as
+well. All references to these types in your custom components will need to be replaced with
+`FlexProps`.
+
+🤖 The codemod will handle all of the changes above for you automatically.
+
+> **Note:** If you were consuming `StackProps` previously for a custom component, be sure to change
+> all references to the old `spacing` prop from `StackProps` to the `gap` prop from `FlexProps`.
 
 ## Component Updates
 
-### Buttons
+### Button
+
+**PR:** [#1978](https://github.com/Workday/canvas-kit/pull/1978)
 
-We've changed the default `type` attribute of all our `buttons` to be `type="button"` instead of the
-default `type="submit"`. This is a breaking change if you expected a button to submit by default.
+We've changed the default `type` attribute for all buttons to `type="button"`. Previously, the
+`type` attribute was not being set which resulted in the buttons
+[defaulting to `type="submit"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attributes).
 
-We found an issue where the default action for `form` controls like `input` and `textarea` is to
-click the first `submit` `button` which in some cases is not a `submit` button.
+This affects all buttons which extend `BaseButton`:
 
-An example of this causing issues was our `Modal.CloseIcon` and `Modal.CloseButton`. When the
-`Modal` is wrapped in a `form` element, the close icon would submit the `form`.
```

Comment:
I removed this example with `Modal.CloseIcon` and `Modal.CloseButton` since technically it was fixed in v8 (#1861). It's a great example though (and a clear example of why this fix is so important on a button-wide level) so maybe I should add it back in.